### PR TITLE
refactor: separate task runners into a separate episode

### PR DESCRIPTION
**Pull request by hollandjg**
Move the task runners section into its own episode, and gather all the examples. 

Motivation:
- Since we say "you can use nox or hatch or anything else", it seems cleaner to have the examples in one place rather than spread through the lesson. 
- We also thought it was valuable to have people learn the manual way first, and then give them the tools to make this into an automated thing, rather than going straight for the automation (which introduces an extra level of complexity and additional things which can go wrong).

### CODE REVIEW

File: _episodes/01-environment.md [hunk 1]
```diff
@@ -1,17 +1,14 @@
 ---
-title: "Environments and task runners"
-teaching: 10
-exercises: 10
+title: "Environments"
+teaching: 5
+exercises: 5
```
Comment (gtdang) [inline _episodes/01-environment.md:4]:
```suggestion
teaching: 10
exercises: 10
```
I think environments is one of the more important concepts for a lot of the attendees this year. Let's leave this at 10m.

File: _episodes/08.01-task-runners.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+more flexible, simpler to understand, specialized to one language,
+general to many languages, and so on.
+
+Examples we'll cover include:
+- [nox][] (Python semi-general), and
+- [hatch][] (specialized for Python package management).
+
+Other examples are:
+- [make][] (fully general),
+- [rake][] (Ruby general),
```
Comment (gtdang) [inline _episodes/08.01-task-runners.md:33]: Let's remove rake. Ruby seems out of context. We could mention that there are many other task-runners for different languages.

File: _episodes/08.01-task-runners.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+> Nox is really just doing the same thing we would do manually (and printing all the steps except the
+> exact details of creating the virtual environment. You can see the virtual environment in `.nox/tests`!
+> How would you activate this environment?
+>
+> > ## Solution
+> > ```bash
+> > . .nox/tests/bin/activate
+> > ```
+> {:.solution}
+{:.challenge}
```
Comment (gtdang) [inline _episodes/08.01-task-runners.md:293]: This section should be moved up after the test and before the docs sections.

**Review by gtdang — CHANGES_REQUESTED** [review]
Looks good John. Just some minor changes/suggestions.

File: _episodes/08.01-task-runners.md (hunk 2 above)
Comment (hollandjg) [reply _episodes/08.01-task-runners.md:33]: Done

**Review by gtdang — APPROVED** [review]
Looks good!

**Pull request merged**